Source/Core/DolphinQt/SkylanderPortal/SkylanderModifyDialog.cpp

```diff
@@ -0,0 +1,337 @@
+    }
+    else if (!edit_last_reset->hasAcceptableInput())
+    {
+      QMessageBox::warning(this, tr("Incorrect last reset time!"),
+                           tr("Make sure that the last reset datetime value is valid!"),
+                           QMessageBox::Ok);
+    }
+    else if (!edit_last_placed->hasAcceptableInput())
+    {
+      QMessageBox::warning(this, tr("Incorrect last placed time!"),
+                           tr("Make sure that the last placed datetime value is valid!"),
+                           QMessageBox::Ok);
+    }
+    else
+    {
+      m_allow_close = true;
+      m_figure_data.skylander_data = {
+          .money = edit_money->text().toUShort(),
+          .hero_level = edit_hero->text().toUShort(),
+          .playtime = edit_playtime->text().toUInt(),
+          .last_reset = {.minute = static_cast<u8>(edit_last_reset->time().minute()),
+                         .hour = static_cast<u8>(edit_last_reset->time().hour()),
+                         .day = static_cast<u8>(edit_last_reset->date().day()),
+                         .month = static_cast<u8>(edit_last_reset->date().month()),
+                         .year = static_cast<u16>(edit_last_reset->date().year())},
+          .last_placed = {.minute = static_cast<u8>(edit_last_placed->time().minute()),
+                          .hour = static_cast<u8>(edit_last_placed->time().hour()),
+                          .day = static_cast<u8>(edit_last_placed->date().day()),
+                          .month = static_cast<u8>(edit_last_placed->date().month()),
+                          .year = static_cast<u16>(edit_last_placed->date().year())}};
+
+      std::u16string nickname = edit_nick->text().toStdU16String();
+      nickname.copy(reinterpret_cast<char16_t*>(m_figure_data.skylander_data.nickname.data()),
+                    nickname.length());
+
+      m_figure->SetData(&m_figure_data);
+      this->accept();
+    }
+  });
+}
+
+void SkylanderModifyDialog::PopulateTrophyOptions()
+{
+  static constexpr u16 KAOS_TROPHY_ID = 3503;
+  static constexpr u16 SEA_TROPHY_ID = 3502;
+
+  if (m_figure_data.figure_id == KAOS_TROPHY_ID)
+  {
+    m_should_show = false;
+    QMessageBox::warning(this, tr("Can't edit villains for this trophy!"),
+                         tr("Kaos is the only villain for this trophy and is always unlocked. No "
+                            "need to edit anything!"),
+                         QMessageBox::Ok);
+    return;
+  }
+
+  auto* hbox_villain_1 = new QHBoxLayout();
+  auto* label_villain_1 = new QLabel(tr("Captured villain %1:").arg(1));
+  auto* edit_villain_1 = new QCheckBox();
+  edit_villain_1->setChecked(m_figure_data.trophy_data.unlocked_villains & 0b1);
+  hbox_villain_1->addWidget(label_villain_1);
+  hbox_villain_1->addWidget(edit_villain_1);
+  m_layout->addLayout(hbox_villain_1);
+
+  u8 shift_count = 1;
+  if (m_figure_data.figure_id != SEA_TROPHY_ID)
+    shift_count++;
```

Comment:
If you want you could make this section less repetitive with something like this:

```suggestion
constexpr size_t MAX_VILLAINS = 4;
std::array<int, MAX_VILLAINS> shift_distances = (m_figure_data.figure_id == SEA_TROPHY_ID) ? {0, 1, 2, 4} : {0, 2, 3, 4};
std::array<QCheckBox*, MAX_VILLAINS> edit_villains;
using unlocked_villains = m_figure_data.trophy_data.unlocked_villains;

for (size_t i = 0; i < MAX_VILLAINS; ++i)
{
  edit_villains[i] = new QCheckBox();
  edit_villains[i]->setChecked(static_cast<bool>(unlocked_villains & (0b1 << shift_counts[i])));
  auto* const label = new QLabel(tr("Captured villain %1:").arg(i + 1));
  auto* const hbox = new QHBoxLayout();
  hbox->addWidget(label);
  hbox->addWidget(edit_villains[i]);
}
```